src/backoff.ml

```diff
@@ -0,0 +1,17 @@
+type t = int * int ref
+
+let k = Domain.DLS.new_key Random.State.make_self_init 
+
+let create ?(max = 32 * 4096) () = (max, ref 1)
+
+let once (maxv, r) =
+  let t = Random.State.int (Domain.DLS.get k) !r in
+  r := min (2 * !r) maxv;
+  if t = 0 then ()
+  else begin
+      for _ = 1 to t do
+        Domain.cpu_relax ()
```

Comment:
Backoff module looks good to me. I thought the implementation of relax is different, but it's doing what's desired (pause). Feel free to mark as resolved.

Reply:
perhaps it's worth adding an `.mli` and stating that Backoff busy-waits but in a way that's nicer to ocaml runtime and system than a simple `while .. do () done`